File: Data Structures & Algorithms/two-integer-sum/submission-6.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        num_cache = {}
        for i in range(len(nums)):
            difference = target - nums[i]
            if difference in num_cache:
                return [num_cache[difference], i]
            else:
                num_cache[nums[i]] = i
        return []

File: Data Structures & Algorithms/binary-search/submission-6.cpp
class Solution {
public:
    int search(const vector<int>& nums, int target) {
        int l = 0, r = nums.size() - 1;
        while (l < r) {
            int m = l + (r - l) / 2;
            if (nums[m] >= target) {
                r = m;
            } else {
                l = m + 1;
            }
        }
        return (l < nums.size() && nums[l] == target) ? l : -1;
    }
};
